### ISSUE TITLE
Cmake 3.26 New Warning/Nag Message

### ISSUE THREAD

**Issue by lwvmobile** [lwvmobile/dsd-fme#107]
https://github.com/lwvmobile/dsd-fme/blob/f3c0199b321ebf3cd65b8379abc04c21f6b8c00d/CMakeLists.txt#L2

Cmake 3.26.1 (on Arch) now presents new nag message to users:

```
CMake Warning (dev) at CMakeLists.txt:1 (project):
  cmake_minimum_required() should be called prior to this top-level project()
  call.  Please see the cmake-commands(7) manual for usage documentation of
  both commands.
This warning is for project developers.  Use -Wno-dev to suppress it.
```

This will be fixed in a future commit after some VM testing in different environments to ensure no breakage. The project line should follow the minimun requirements line now.

```
set(CMAKE_LEGACY_CYGWIN_WIN32 0) #annoying warning in cygwin
#Ubuntu 18.04 uses 3.10.2 currently -- almost EOL
#Ubuntu 20.04 uses 3.16.3
#Ubuntu 22.04 uses 3.22.1
#Arch currently on 3.26.1 -- new nag warning starts on this one, see below
#cmake_minimum_required() should be called prior to this top-level project()
#Cygwin currently using 3.23.2 (or newer)
cmake_minimum_required(VERSION 3.10.2)
project(dsd-fme)
```

**Comment by lwvmobile** [comment]
Apparently, I've discovered that moving the line
project(dsd-fme)`
underneath the 
`set(CMAKE_LEGACY_CYGWIN_WIN32 0)`
line causes another warning message, and that removing the line entirely fixes that warning, making the legacy line obsolete now, whereas before, it would warn if the line wasn't set, and would warn about the requirement numbering, but setting it below wouldn't fix it. Seems that cmake and some environments (cygwin) are just picky about the ordering of the lines. I've removed the legacy line and now cygwin doesn't complain at all during cmake, guess it just didn't like the ordering of things so well.
